### ISSUE TITLE
Bug report: Application crashed with AttributeError

### ISSUE THREAD

**Issue by giovannipizzi** [aiidalab/aiidalab-qe#866]
To reproduce: click on "Job History" twice (second time when already in that page view)

## Automated report

_This issue was created with the app's automated bug reporting feature.
Attached to this issue is the full traceback as well as an environment
fingerprint that contains information about the operating system as well as all
installed libraries._

## Additional comments (optional):

_Example: I submitted a band structure calculation for Silica._

## Attachments

<details>
<summary>Traceback</summary>

```python-traceback
/opt/conda/lib/python3.9/contextlib.py in __exit__(self, typ, value, traceback)
    124         if typ is None:
    125             try:
--> 126                 next(self.gen)
    127             except StopIteration:
    128                 return False

/opt/conda/lib/python3.9/site-packages/traitlets/traitlets.py in hold_trait_notifications(self)
   1500                 for changes in cache.values():
   1501                     for change in changes:
-> 1502                         self.notify_change(change)
   1503 
   1504     def _notify_trait(self, name, old_value, new_value):

/opt/conda/lib/python3.9/site-packages/ipywidgets/widgets/widget.py in notify_change(self, change)
    692                 # Send new state to front-end
    693                 self.send_state(key=name)
--> 694         super(Widget, self).notify_change(change)
    695 
    696     def __repr__(self):

/opt/conda/lib/python3.9/site-packages/traitlets/traitlets.py in notify_change(self, change)
   1515     def notify_change(self, change):
   1516         """Notify observers of a change event"""
-> 1517         return self._notify_observers(change)
   1518 
   1519     def _notify_observers(self, event):

/opt/conda/lib/python3.9/site-packages/traitlets/traitlets.py in _notify_observers(self, event)
   1562                 c = getattr(self, c.name)
   1563 
-> 1564             c(event)
   1565 
   1566     def _add_notifiers(self, handler, name, type):

/opt/conda/lib/python3.9/site-packages/aiidalab_qe/app/wrapper.py in _on_job_history_toggle(self, change)
     78             ]
     79         else:
---> 80             self._view.main.children = [self._view.app]
     81 
     82     def _set_event_handlers(self) -> None:

AttributeError: 'AppWrapperView' object has no attribute 'app'
```
</details>

<details>
<summary>Environment fingerprint</summary>
<pre>eNptWNuS47gN_ZUu53Xlsi52T8_bZt62ZpJJejZbqVTKRUm0xDZFsknKHvXW_HtAALTVU3k0eBEIHBwc-
M_NRfqgrNl8LH_ZOC3iyfpp8_HPjfDdqKLs4uzl5uN_NoemVXHzy2bzX9i3xNGa4-3opt4-bcsaVu-mv3yoit_b2cT54fnL14cv1
jw8S_dQPZTlx_rwsW4efv_26aHaVc3mB9wourMYZMAvK9WLQkQ7qRBVB3fttuV2J3ZwP611Nvm0qbaHbXUzjouT_nWWs8QT1fa-X
01u9iouYYYtndS6s-YCu-DW1a5pRsffGx081LoF7eW2uduDnHuL5v1q--ss4MWTDM7LENJ6A07e1y-q9TZ09xvL29JVGKOkf9rhy
9bu31aKq_Xnk7bXgHvq-x4t2mRqIA9r23G0k_x_C8V0gjCGDi_aVfW23O3q9cFXPlZCMES5Pvkqc6Rgafvz0k8PfL_heFX9IGM4t
iLI_ARBKbSFU2cBxsP2Ax8CJL4iuGoyaDm1iIcybzHGRhFlX0RIfkCXHumdJqhqjJPG3U8EE7Moi_cd6LRzvfIBdzSUWucm2xN-P
tA9qQousnAijmhu6Co_WFMV3emk0jPKnIq7uWiV6ZUZAq1zOvHV9baqyIEQoz1LQ9mET6JtMd1xkEZ6qABPKMGzMaKv6Sxe9lfRS
o0nIUcpdy3UUCe0XsE_mZz1MWxPs-mitToU2s9FJ7pR4t0HfHjb-cVFRCsflGKO6jTrYGfX4EJJYG21hMNg2XNyW63MWZKn-_Qbn
jSiXw255W3UCvGQs_Mpff6TNdFb8rakDKNb6CW70qAxvxw2YfQ76cE3RdBtIFP1Llv5MxRfTg_4veffwwXj36AX3Sh8kLEwgDOh1
Ru-oeKa7JTvZkJUSVDotOrOYADobR_z7yI4ZQyeTCSFmeqstl5MgvGyR5PpBSWP6hANBbNeMQrTQxQHBitviHb2XEYEL0ySHbxwY
zLX4BdFaOl0doFe3st2HvjsAbPQSyBNRtSe39jL0xxk_33SnBsyAnVBzInFM9h6Fc5H275APwiR2Le8L0F-W7wC-kCymEAdAi-
glPe2C9HDG48uRZ29PaSVWV9VR9XFhSGh0qWJXHQNM6ychNLHC6Sq54dkmoS9EBqrTKSMpaDIOVE300WNuJHfO-nSwwYPqMaFihd
kB3DnBFBwTlCdL8GaAJDEXIJr1G5OSkttEQt1suFureKR21LNkTxBCu9g3jdknKmh3toM8GHBPAaeMeLwO5joi3qjyFCND15Ko2V
kR5Np3GOi65z7cZ6EOXklAVOIAPoOmgHkuZaTTfUQOXVauJeWaDKCaiT9mACdyJnA1NTGoJUCn0C-i0lGAYkgyuaP3Feh_QF8OyT
l_W3ZLcAUBmnrkELHRgtZmUS_btzqhqDHbd2sTQWQIyRLh592Q2TYUpGFmw1e8UjvuRkL-T1CgGS_yriKoRdmkACOsALXi-xV5oG
07TdlXqBnYcTxUy-2JfyX3B0SbPZ45Amr_x2Mmvzulxmclv4IPAJZoIfS9rxAcLoxL9sLqJ4L82359G4lYuFALWTNdF8b5xZd3q2
NpBoSH354ZyzcMqQCDKuA3lePNweqxKw_Ld6jXmZCPEPnViiWMjOf1fXeER7ZEKzO72rQqIU_r6oUxClUMPTnbvUSLd6Wo7YAHn8
rfcjGJKZWMAES-X8RZ8uXparBDe-W_bm3VxKzj2yY3bM4yVVXmER02jLd1awr7sZCGaBxydFP5JYk7GwkN_jk3ASssOTW9Uita3I
QcT9a7hi4zU3Efpn8pzCkdrEi3mmWwa_4Ypr1qWNZ1tDvqHrVRao2PGPaTosQWFVTc0o2hmBu0GAChQwNFWFJOTNtGg1ExIImsjA
yAHklxaKyFD6gOSah-p20Du0c9EXJKycuoc1AIiWrcEIxcKBsrT2ju4D6amUrwqimFarNPHF2K1IwFvTKmPNC31yxSiLb5Bk1XCL
6iujTAThjFCBrWGqScZJarFgaDK9uJboddGyRWaJmg-2gNUAZhNVYAN1OTOpMHFpW2RYsi0w8K7872ZH8Iq3hhjBzuaTH3YcQ-
BzfC-JDywASRnLe9mjVMB2kz2fHvypObIVfUo5kfYqtOw-JR4-JqqHyjkZMkgGHZZunQRbJoOYohgnrC6Kgop7mNGcDvkLayGluS
bjsJXDZNCl6IalG52EyiaOcQ7FCX9ncFl0sUvM846kEmhpXQnIZP07F58ICw8ZQYWyeUH5kU5Lgwi_rFRgTvIWmtB4VYDSUhQRZs
SI7t7TANWSg5vN1-aRO__wDX9BQHOArQZDfpLHAchM3cE9JptlrrKKGiMYt0GRorK0Y-Nl0vM21FcOQF4DyY8zDBM-TX-8UXSX9T
Lbf_viGBgbJAvUK7TLTzYETy2bQHUIvQYUClGBSdndaKp_wui_L8z8-cyer0fI38Umv4b383Unz_Px5PZm4JYWBqixlDkO_gCXEW
6Kpl35dnkFIUb08ciywx4OwkJzpCmqiWq0AI9yG99WJ1GQLbYeBw39b9GJhbdqQ4Y2eWfKL_v3rl89IOTv-zBtOnVVGKgy1WcVlS
eDlK3Avx74mAsi21WxFNOTnRCbbRbDMLh8JGiv7tssCukJo0VKRloo8OuxSr_2Qvh46BVVRoDhbQxTs-W8FGqeClGeXu8gTFmaQE
_XYmp_7DCqo-uZFGHmuxptknF1-9GG_paoMI2i7eVZ9Dn56dlDf6ZOEymAUDF121bfSABmUvOR5v9o6GyLe5wZ6dsUDGWDtV5100
sIZrFPGQgTSLlhp7rhdh2W6vzX5AQw-J4HAYE3XAU5Ep7DZPjEbAjlPQAqsRdjnOHopegfP7aLG0BC2QRmfktb_ac6EQly6EKqVc
ow255aCmn6rFQpTKN_yfIZ3WM9e0EgRX_uJ_iuiUANoVdQko_b5OYtL4xMK18CjQuqSdMEbBwg9TZY5FcRafM8mTcgC-oFCVV4z1
83udIz2yNuzZgXWug_GrMlqskPS6tx6keD_JUD1LyslclEAFOirWKcwuVTEtBerAVazzXM1aceriN3Y22HVHK4dqEj-w6VCd66yl
Qa0H5NghgGYAxAIjFC3JlIyWq-jlJSUmpiFhOnRSa9sUkW3ll-
uloNpbwHmv4rQRxjF4hrnMDOvBkNyZhGeqJH1-ptyjmXibvPjx_8AOkcTiw==</pre>
</details>

**By submitting this issue I confirm that I am aware that this information can
potentially be used to determine what kind of calculation was performed at the
time of error.**

**Comment by giovannipizzi** [comment]
Also: when going directly to the Job list from the home page, it crashes:
```python
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
/tmp/ipykernel_8560/1471599892.py in <cell line: 4>()
      2 
      3 qi = QueryInterface()
----> 4 qi.filters_layout

AttributeError: 'QueryInterface' object has no attribute 'filters_layout'
```

**Comment by giovannipizzi** [comment]
And when I click "inspect", now it just opens the page for a new workflow, so I can't check anymore my old calculations...

**Comment by superstar54** [comment]
Hi @giovannipizzi , thanks for reporting the issue.

> Also: when going directly to the Job list from the home page, it crashes:

Yes, I can confirm this. I forgot to update it. Will fix it soon.


For the other two issues: 1) click on "Job History" twice, 2) click "inspect", now it just opens the page for a new workflow; I can not reproduce them.

@mikibonacci could you try and see if you can reproduce them?

**Comment by superstar54** [comment]
The `Job history` from the home page should be fixed by PR #867 .
